src/mesh/mesh.cxx

```diff
@@ -484,7 +492,21 @@ void Mesh::addRegion3D(const std::string &region_name, const Region<> &region) {
     throw BoutException(_("Trying to add an already existing region {:s} to regionMap3D"),
                         region_name);
   }
-  regionMap3D[region_name] = region;
+
+  int id = -1;
+  for (size_t i = 0; i < region3D.size(); ++i) {
+    if (region3D[i] == region) {
+      id = i;
+      break;
+    }
+  }
```

Comment:
This is `std::find_if` :)

I think this whole bit might be `try_emplace`, but that's C++17